Sources/Clickstream/Core/Interface/ClickStream.swift

```diff
@@ -108,12 +226,21 @@ public final class Clickstream {
     /// - Parameters:
     ///   - networkConfiguration: Network Configurations needed for connecting socket
     ///   - constraints: Clickstream constraints passed from the integrating app.
+    ///   - dataSource: ClickStreamDataSource instance passed from the integrating app.
     ///   - eventClassification: Clickstream event classification passed from the integrating app.
+    ///   - request: URL request with secret code for the services to authenticate.
+    ///   - appPrefix: Application name without any space or special characters that needs
+    ///   to be passed from integrating app.
     /// - Returns: returns a Clickstream instance to keep throughout the project.
     ///            You can always get the instance by calling getInstance()
-    @discardableResult public static func initialise(networkConfiguration: NetworkConfigurations,
-                                                     constraints: ClickstreamConstraints? = nil,
-                                                     eventClassification: ClickstreamEventClassification? = nil) throws -> Clickstream? {
+    @discardableResult public static func initialise(with request: URLRequest,
```

Comment:
With JSONString developers can make typo mistakes, hence instead of JSONString we preferred ClickstreamConstraints instance